public/app/plugins/datasource/prometheus/language_provider.ts

```diff
@@ -443,8 +445,8 @@ export default class PromQlLanguageProvider extends LanguageProvider {
     // The rounding may seem strange but makes relative intervals like now-1h less prone to need separate request every
     // millisecond while still actually getting all the keys for the correct interval. This still can create problems
     // when user does not the newest values for a minute if already cached.
-    params.set('start', this.roundToMinutes(tRange['start']).toString());
-    params.set('end', this.roundToMinutes(tRange['end']).toString());
+    params.set('start', roundSecToMin(tRange['start']).toString());
+    params.set('end', roundSecToMin(tRange['end']).toString());
```

Comment:
I am bit confused here. This code does not seem to be different which means we already did cache the query for 1 minute right? So the multiple queries should not happen even before even if there was a few second difference. Or is this different query from the one that was duplicated?

Reply:
This is `fetchSeriesLabels` function (`/api/v1/series?...`) that caches the labels for specific metrics. So If you metric is for example `go_gc_duration_seconds` it caches all the label-values associated with this metric. 

What this PR does -> it makes sure that we are not fetching metrics too often. I have decided to solve this but not refreshing the metric if they are within the same 1 minute.

Reply:
Ah ok so this is a different API call, makes sense then.